### DISCUSSION TITLE
Duration for native transitions...does it exist?

### DISCUSSION THREAD

**Discussion by Aryk** [software-mansion/react-native-screens#1114] [Q&A] [answered]
According to ios you can tweak the duration of the transition:

https://stackoverflow.com/a/28767169/7180620

I'm using react native navigation v5 and I can't figure out how to adjust the transition time between screens.

Is this possible to do? Probably something in screenOptions, but I'm not seeing any documentation for it.

Expo: SDK 41 Managed
React-native-navigation v5.0

**Comment by WoLewicki** [comment]
If you mean the `default` iOS transition, it cannot be changed from code. Other options can be adjusted in this file e.g. here: https://github.com/software-mansion/react-native-screens/blob/master/ios/RNSScreenStackAnimator.m#L142 and here: https://github.com/software-mansion/react-native-screens/blob/master/ios/RNSScreenStackAnimator.m#L152 for `fade` transition, by changing the code to something like `[UIView animateWithDuration:(your_value)`

**Reply by Aryk** [reply]
I am using the standard ios push transition. I want to make it more snappy like you see on IG. 

I don't know any objective C, could I just use patch package for this?  

Is it work putting in a feature request for this? Seems like it should be a setting that should be exposed.

**Reply by WoLewicki** [reply]
As I said, the default transition is not customizable. You can use `simple_push` transition, which tries to imitate the default one but without shadow and specific header transition. You can use patch package to apply changes to the files.

**Reply by Aryk** [reply]
Got it I think. So the RNSScreenStackAnimator is connected to simple push, so I simply switch to simple push and then replace [this line ]([UIView animateWithDuration:(your_value))with something like:

```
[UIView animateWithDuration:500
```

For 500ms simple_push animation.

Is that right?

**Reply by WoLewicki** [reply]
Yes, that should be all you need. Just remember to switch it in both `pop` and `push` like in my initial comment.

**Reply by taitran2512** [reply]
Is there any way down duration for android, this is my screenOptions, on IOS it's ok, but on Android duration transition is take to much time
const screenOptions = {
	stackAnimation: 'slide_from_right',
	stackPresentation: 'transparentModal',
	replaceAnimation: 'push',
}

**Reply by WoLewicki** [reply]
@taitran2512 you can see comments in this issue: https://github.com/software-mansion/react-native-screens/issues/489 to see how to change the duration on `Android`.

**Reply by hirbod** [reply]
@WoLewicki is there any chance you could polish simple_push a bit more to make it feel more like the default iOS one? This one is so different in performance while swiping (feels janky/jerky) and the spring/snaps when you release your finger are different as well. I don't mind the missing header animation, but the missing backdrop shadow.

The only reason to use simple_push is for fullScreenSwipe gesture, which I won't for the reasons.

**Reply by WoLewicki** [reply]
There shouldn't be any performance differences when using `simple_push` and I have no idea how to change it then. As for the missing header animation and backdrop shadow, the first one cannot be recreated and the second one was omitted since this animation option was introduced in order to provide the simplest transition between screens. As for the snapping, I have no knowledge how to at the same time provide the 'sticking to the finger' transition progress, and spring at the finger releasing.

**Reply by Aryk** [reply]
@WoLewicki I was able to change the number in the RNStackNavigator, but there is a weird behavior. The new duration only starts taking effect once I use a swipeGesture to go back one screen.  Then moving forward it will use that new time whenever I navigate to that Screen. 

Do you have any idea what could be causing that?

Until then I believe it is using some default value. I even changed this:

```
- (instancetype)initWithOperation:(UINavigationControllerOperation)operation
{
  if (self = [super init]) {
    _operation = operation;
    _transitionDuration = 0.1; // default duration
  }
  return self;
}
```

And it still wouldn't use the 0.1 value until I did a swipe back gesture.

**Reply by WoLewicki** [reply]
@Aryk can you provide a simple repo with the changes you made applied with e.g. [patch-package](https://www.npmjs.com/package/patch-package)? It will be easier to debug it then.

**Reply by Aryk** [reply]
Hey @WoLewicki - I still didn't figure this out, but I had to change gears and I'm not messing with code at the moment. As I onboard engineers to Tribefy, I'll have them take another look at this. 👍

**Reply by WoLewicki** [reply, marked as answer]
We added `transitionDuration` on `iOS`: https://github.com/software-mansion/react-native-screens/blob/7c6e14b45834cf009c7069d46a528cf792d2a15e/src/types.tsx#L252, it should be ready to use. Is it would you need?
